sycl/plugins/unified_runtime/ur/adapters/level_zero/usm.cpp

```diff
@@ -12,8 +12,61 @@
 
 #include "context.hpp"
 #include "event.hpp"
+#include "usm.hpp"
+
 #include "ur_level_zero.hpp"
 
+#include <umf_helpers.hpp>
+
+ur_result_t umf2urResult(umf_result_t umfResult) {
+  if (umfResult == UMF_RESULT_SUCCESS)
+    return UR_RESULT_SUCCESS;
+
+  switch (umfResult) {
+  case UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY:
+    return UR_RESULT_ERROR_OUT_OF_HOST_MEMORY;
+  case UMF_RESULT_ERROR_MEMORY_PROVIDER_SPECIFIC: {
+    auto hProvider = umfGetLastFailedMemoryProvider();
+    if (hProvider == nullptr)
+      return UR_RESULT_ERROR_UNKNOWN;
+
+    ur_result_t Err = UR_RESULT_ERROR_UNKNOWN;
+    umfMemoryProviderGetLastNativeError(hProvider, nullptr,
+                                        reinterpret_cast<int32_t *>(&Err));
+    return Err;
+  }
+  case UMF_RESULT_ERROR_INVALID_ARGUMENT:
+    return UR_RESULT_ERROR_INVALID_ARGUMENT;
+  case UMF_RESULT_ERROR_INVALID_ALIGNMENT:
+    return UR_RESULT_ERROR_UNSUPPORTED_ALIGNMENT;
+  case UMF_RESULT_ERROR_NOT_SUPPORTED:
+    return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+  default:
+    return UR_RESULT_ERROR_UNKNOWN;
+  };
+}
+
+usm::DisjointPoolAllConfigs InitializeDisjointPoolConfig() {
+  const char *PoolConfigVal = std::getenv("UR_L0_USM_ALLOCATOR");
```

Comment:
I'm not sure about this. All adapters will use UMF, yes. However, each adapter will still use a separate instance of a UMF memory pool. If you have an application that uses L0 and CUDA or HIP adapter, it might make sense to configure pooling for each adapter separately.

Also, UR_L0_USM_ALLOCATOR is an experimental variable and I think we should keep it this way for now.
We have this structure that can be used to configure a memory pool when calling `urUSMPoolCreate` : https://oneapi-src.github.io/unified-runtime/core/api.html?highlight=limits#_CPPv425ur_usm_pool_limits_desc_t Ideally we would be able to specify all required pooling properties through that structure but that will require optimizing the disjoint pool (or adding new, generic properties - the ones in UR_L0_USM_ALLOCATOR  are specific to the current pool implementation).

Reply:
> If you have an application that uses L0 and CUDA or HIP adapter, it might make sense to configure pooling for each adapter separately.

that is a good answer. Thanks @igchor .